nmigen_boards/genesys2.py

```diff
@@ -0,0 +1,390 @@
+        Resource("clk", 0, DiffPairs(p="AD12 ", n="AD11", dir="i"),
+                 Clock(200e6), Attrs(IOSTANDARD="LVDS")),
+        Resource("rst", 0, PinsN("R19", dir="i"),
+                 Attrs(IOSTANDARD="LVCMOS33")),
+        *LEDResources(pins="T28 V19 U30 U29 V20 V26 W24 W23",
+                      attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("fan", 0,
+                 Subsignal("pwm", Pins("W19", dir="o")),
+                 Subsignal("tach", Pins("V21", dir="i")),
+                 Attrs(IOSTANDARD="LVCMOS33")),
+        UARTResource(0, rx="Y20", tx="AE20",
+                     attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("i2c", 0,
+                 Subsignal("scl", Pins("AE30", dir="io")),
+                 Subsignal("sda", Pins("AF30", dir="io")),
+                 Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("ddram", 0,
+                 Subsignal("a", Pins(
+                     "AC12 AE8 AD8 AC10 AD9 AA13 AA10 AA11 "
+                     "Y10 Y11 AB8 AA8 AB12 AA12 AH9 AG9", dir="o"),
+                     Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("ba", Pins("AE9 AB10 AC11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("ras_n", Pins("AE11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("cas_n", Pins("AF11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("we_n", Pins("AG13", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("dm", Pins("AD4 AF3 AH4 AF8", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("dq", Pins(
+                     "AD3 AC2 AC1 AC5 AC4 AD6 AE6 AC7 "
+                     "AF2 AE1 AF1 AE4 AE3 AE5 AF5 AF6 "
+                     "AJ4 AH6 AH5 AH2 AJ2 AJ1 AK1 AJ3 "
+                     "AF7 AG7 AJ6 AK6 AJ8 AK8 AK5 AK4", dir="io"),
+                     Attrs(IOSTANDARD="SSTL15_DCI", ODT="RTT_40")),
+                 Subsignal("dqs",
+                           DiffPairs(p="AD2 AG4 AG2 AH7",
+                                     n="AD1 AG3 AH1 AJ7", dir="io"),
+                           Attrs(IOSTANDARD="DIFF_SSTL15_DCI",
+                                 ODT="RTT_40")),
+                 Subsignal("clk",
+                           DiffPairs(p="AB9", n="AC9", dir="o"),
+                           Attrs(IOSTANDARD="DIFF_SSTL15_DCI")),
+                 Subsignal("cke", Pins("AJ9", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("odt", Pins("AK9", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("reset", PinsN("AG5", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15")),
+                 Attrs(SLEW="FAST", OUTPUT_IMPEDANCE="RDRV_40_40")),
+        Resource("audio", 0,
```

Comment:
There's no single standardized "audio" interface. This looks like it's a combination of I2C, I2S, and something else? The resource should probably be split then.

Reply:
Split the resources:
* `audio_i2c`, ADAU1761 I2C, control port
* `audio_i2s`, ADAU1761 ADC/DAC, serial data ADC, DAC, [https://en.wikipedia.org/wiki/I²S](https://en.wikipedia.org/wiki/I%C2%B2S)
* `audio_clk`